backend/src/schema/resolvers/searches/queryString.js

```diff
@@ -0,0 +1,47 @@
+export function queryString(str) {
+  const normalizedString = normalizeWhitespace(str)
+  const escapedString = escapeSpecialCharacters(normalizedString)
+  return `
+${matchWholeText(escapedString)}
+${matchEachWordExactly(escapedString)}
+${matchSomeWordsExactly(escapedString)}
+${matchBeginningOfWords(escapedString)}
+`
+}
+
+const matchWholeText = (str, boost = 8) => {
+  return `"${str}"^${boost}`
```

Comment:
> [<img alt="mattwr18" height="40" width="40" align="left" src="https://avatars1.githubusercontent.com/u/38261864?s=88&v=4">](https://github.com/mattwr18) **Authored by [mattwr18](https://github.com/mattwr18)**
_<time datetime="2020-03-17T09:46:48Z" title="Tuesday, March 17th 2020, 10:46:48 am +01:00">Mar 17, 2020</time>_

---
> **Outdated (history rewrite)** - original diff
---
```diff
@@ -0,0 +1,53 @@
+export function queryString(str) {
+  const normalizedString = normalizeWhitespace(str)
+  const escapedString = escapeSpecialCharacters(normalizedString)
+  return `
+${matchWholeText(escapedString)}
+${matchEachWordExactly(escapedString)}
+${matchSomeWordsExactly(escapedString)}
+${matchBeginningOfWords(escapedString)}
+`
+}
+
+const matchWholeText = (str, boost = 8) => {
+  return `"${str}"^${boost}`
```


thank you for giving the docs so that your reviewers can understand how the boost works!
I didn't really understand how you came to the defaults. Maybe you could explain?

Reply:
> [<img alt="Mogge" height="40" width="40" align="left" src="https://avatars1.githubusercontent.com/u/38261864?s=88&v=4">](https://github.com/Mogge) **Authored by [Mogge](https://github.com/Mogge)**
_<time datetime="2020-03-17T11:16:14Z" title="Tuesday, March 17th 2020, 12:16:14 pm +01:00">Mar 17, 2020</time>_

---


I thought: 1, 2, 4, 8
Next would be 16
Matching the whole query exactly -> 8
Matching all words exactly -> 4
Matching at least one word exactly -> 2
Match the beginning of some words -> 1